rxnetty/src/main/java/io/reactivex/netty/protocol/http/UnicastContentSubject.java

```diff
@@ -138,7 +137,14 @@ private UnicastContentSubject(final State<T> state, long noSubscriptionTimeout,
      */
     public boolean disposeIfNotSubscribed() {
         if (state.casState(State.STATES.UNSUBSCRIBED, State.STATES.DISPOSED)) {
-            state.bufferedSubject.subscribe(Subscribers.empty()); // Drain all items so that ByteBuf gets released.
+            state.bufferedSubject.subscribe(new Action1<T>() {
```

Comment:
To make it safer for future issues like this, I think it will be better to split the subject into an `Observable` and `Observer` with `Observable` always wired with auto-release. Something like:

```java
        private final Observer<T> bufferedObserver;
        private final Observable<T> bufferedObservable;

        public State(Action0 onUnsubscribe) {
            this.onUnsubscribe = onUnsubscribe;
            final BufferUntilSubscriber<T> bufferedSubject = BufferUntilSubscriber.create();
            bufferedObservable = bufferedSubject.lift(new AutoReleaseByteBufOperator<T>());
            bufferedObserver = bufferedSubject;
        }
```
and then we do not store the `Subject` variable.